addons/binding/org.openhab.binding.kvv/src/main/java/org/openhab/binding/kvv/internal/KVVHandler.java

```diff
@@ -0,0 +1,226 @@
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        this.handleRefresh(channelUID);
+    }
+
+    /**
+     * Refreshes all {@link Channel Channels}
+     */
+    public void refreshAll() {
+        for (final ChannelUID channel : this.channels) {
+            this.handleRefresh(channel);
+        }
+    }
+
+    /**
+     * Refreshes a specific {@link Channel}
+     *
+     * @param channelUID the {@link ChannelUID} of the {@link Channel}
+     */
+    @SuppressWarnings("null")
+    private void handleRefresh(ChannelUID channelUID) {
+        final int id = Integer.parseInt(channelUID.getId().replaceAll("[^0-9]", ""));
+        final String type = channelUID.getId().substring(channelUID.getId().lastIndexOf('_') + 1);
+
+        logger.debug("Refresh for '" + channelUID.getId() + "'. Id is #" + id + ", type is '" + type + "'");
+
+        if (this.departures == null) {
+            logger.info("departure is null...");
+            updateState(channelUID, new StringType(""));
+            return;
+        }
+
+        if (this.departures.getDepartures().size() - 1 < id) {
+            logger.info("Train with #" + id + " does not exist.");
+            updateState(channelUID, new StringType(""));
+            return;
+        }
+
+        final Departure departure = this.departures.getDepartures().get(id);
+        if (departure == null) {
+            updateState(channelUID, new StringType(""));
+            return;
+        }
+
+        switch (type) {
+            case "name":
+                updateState(channelUID, new StringType(departure.getRoute()));
+                break;
+            case "destination":
+                updateState(channelUID, new StringType(departure.getDestination()));
+                break;
+            case "eta":
+                updateState(channelUID, new StringType(departure.getTime()));
+                break;
+            default:
+                updateState(channelUID, new StringType(""));
+        }
+    }
+
+    /**
+     * Updates the local list of departures.
+     *
+     * @param departures the new list of departures
+     */
+    private synchronized void setDepartures(final DepartureResult departures) {
+        this.departures = departures;
+
+    }
+
+    @Override
+    public void initialize() {
+        logger.error("Start initializing!");
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            logger.info("Starting inital fetch");
+            final UpdateTask updateThread = new UpdateTask();
+            final DepartureResult departures = updateThread.get();
+            if (departures != null) {
+                this.setDepartures(departures);
+                updateStatus(ThingStatus.ONLINE);
+
+                // Schedule update and refresh tasks
+                new Timer().schedule(new UpdateTask(), 0, this.config.updateInterval);
+                new Timer().schedule(new RefreshTask(), 0, this.config.updateInterval);
```

Comment:
First of all: do not create schedulers on your on, re-use existing ones (`scheduler.scheduleWithFixedDelay`). And why don't you update the channel states directly after you got the results? You would only need one task then. This could also be used for the initial refresh. 

If you go the "bridge" path, you could refactor the "refresh" (getting the data and updating the channels) to a method of the thing and call that from the refresh thread within the bridge.